src/app/api/game-sessions/route.ts

```diff
@@ -20,21 +91,280 @@ export async function GET() {
  */
 export async function POST(req: NextRequest) {
   try {
-    const json = await req.json();
+    const user = await getCurrentUser();
+    if (!user) {
+      return new Response("ERROR: Unauthorized request", { status: 401 });
+    }
+    if (user.role != "admin") {
+      return new Response("ERROR: No valid permissions", { status: 403 });
+    }
 
-    const body = insertGameSessionSchema.parse(json);
+    const gameSessionToInsert = insertGameSessionSchema.parse(await req.json());
 
-    if (new Date(body.date) < new Date()) {
+    if (new Date(gameSessionToInsert.date) < new Date()) {
       return new Response("Date cannot be in the past", { status: 400 });
     }
 
-    await db.insert(gameSessions).values(body);
+    // Check that a record doesn't already exist for this date
+    const gameSession = await db.query.gameSessions.findFirst({
```

Comment:
variables could be more self-documenting -- e.g. if this was called `existingGameSession` it makes it really clear what the variable is used for checking